src/wp-includes/author-template.php

```diff
@@ -229,13 +229,29 @@ function the_author_meta( $field = '', $user_id = false ) {
  */
 function get_the_author_link() {
 	if ( get_the_author_meta( 'url' ) ) {
-		return sprintf(
+		global $authordata;
+
+		$author_url          = get_the_author_meta( 'url' );
+		$author_display_name = get_the_author();
+
+		$author_url_link_html = sprintf(
 			'<a href="%1$s" title="%2$s" rel="author external">%3$s</a>',
-			esc_url( get_the_author_meta( 'url' ) ),
+			esc_url( $author_url ),
 			/* translators: %s: Author's display name. */
-			esc_attr( sprintf( __( 'Visit %s&#8217;s website' ), get_the_author() ) ),
-			get_the_author()
+			esc_attr( sprintf( __( 'Visit %s&#8217;s website' ), $author_display_name ) ),
+			$author_display_name
 		);
+
+		/**
+		 * Filters the author URL link HTML.
+		 *
+		 * @since 6.0.0
+		 *
+		 * @param string $author_url_link_html    The default rendered author HTML link.
+		 * @param string $author_url              Author's URL.
+		 * @param string $authordata              Author user data.
+		 */
+		return apply_filters( 'author_url_link_html', $author_url_link_html, $author_url, $authordata );
```

Comment:
Filters are often named for their function, in this case `get_the_author_link`. 

Is there a reason you prefer this name?

Reply:
Looking at some other similar functions, we have:
* `the_author` filter in `get_the_author()`
* `the_modified_author` filter in `get_the_modified_author`
* `the_author_posts_link` filter in `get_the_author_posts_link`

So I think `the_author_link` would be the preferred name here for consistency.

Reply:
I'm also not sure about passing the `$author_url` and `$authordata` parameters, as none of the other three filters pass that. However, the `author_link` filter in `get_author_posts_url()` passes the `$author_id` and `$author_nicename` parameters.

I think this might benefit from a separate discussion on whether we want to consistently pass `$authordata` to all of these filters. For now, maybe just passing `$author_url_link_html` would be fine?

Reply:
> For now, maybe just passing `$author_url_link_html` would be fine?

For the filter to be useful, I think a variable identifying the author needs to be passed. I don't have a preference as to whether it's the ID or the user object.